packages/cdktf-cli/src/bin/cmds/handlers.ts

```diff
@@ -440,21 +445,41 @@ export async function providerAdd(argv: {
     }
     return errorMessage;
   }, "");
+
   const needsGet = results.some(
-    (item) => item.status === "fulfilled" && item.value === true
+    (item) =>
+      item.status === "fulfilled" && item.value.addedLocalProvider === true
   );
+  const packagesToInstall = results.reduce((packages, item) => {
```

Comment:
nit: I think changing the `reduce` to a chained `filter` and `flatMap` would make this more readable 😁
#no-reduce 😄